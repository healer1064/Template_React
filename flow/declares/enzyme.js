// @flow

declare module 'enzyme' {

  declare var shallow: Function

}
