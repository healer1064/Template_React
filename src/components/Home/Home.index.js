// @flow
import { connect } from 'react-redux'
import Home from './Home'


type HomeStateProps = {
  init: InitState,
}


function mapStateToProps (state: GlobalReducerState): HomeStateProps {

  const { init } = state

  return {
    init,
  }

}


export default connect(mapStateToProps)(Home)
