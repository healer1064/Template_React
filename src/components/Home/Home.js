// @flow
import React from 'react'
import { Link } from 'react-router'
import css from './Home.style.css'

import img from '../../styles/images.css'
import sprites from '../../styles/sprites.css'


const Home = (): React$Element<any> => {

  return (
    <div className={`${css.home} flex flex-column justify-center items-center pt7 tc`}>
      This is the offline app...
      <ul>
        <li>This is a test item</li>
      </ul>
      <Link to="/profile">
        <div className={img.logo} />
      </Link>
      <div className="pt2">
        <div className={sprites.facebookIcon} />
      </div>
    </div>
  )

}


export default Home
