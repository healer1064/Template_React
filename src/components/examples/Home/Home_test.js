// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'


type OverrideProps = {
  authenticated: any,
  logout: Function,
  history: Object,
}

const mockProps: OverrideProps = {
  authenticated: false,
  logout: () => {},
  history: {
    replace: () => {},
  },
}

it('<Home> renders the Home content', () => {

  const wrapper = shallow(
    <Home
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})


it('<Home> renders the logout button if authenticated', () => {

  const wrapper = shallow(
    <Home
      {...mockProps}
      authenticated
    />
  )

  expect(wrapper).toMatchSnapshot()

})
