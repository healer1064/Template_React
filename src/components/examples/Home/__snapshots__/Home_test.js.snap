// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> renders the Home content 1`] = `
<div
  className="home"
>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="Home"
  />
  This is the app...
  <ul>
    <li>
      This is a test item
    </li>
  </ul>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/profile"
  >
    <div
      className="logo"
    />
  </Link>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/todos"
  >
    <div
      className="facebookIcon"
    />
  </Link>
</div>
`;

exports[`<Home> renders the logout button if authenticated 1`] = `
<div
  className="home"
>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="Home"
  />
  This is the app...
  <ul>
    <li>
      This is a test item
    </li>
  </ul>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/profile"
  >
    <div
      className="logo"
    />
  </Link>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/todos"
  >
    <div
      className="facebookIcon"
    />
  </Link>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
